gradle/wrapper/gradle-wrapper.properties

```diff
@@ -0,0 +1,5 @@
+distributionBase=GRADLE_USER_HOME
```

Comment:
This file is automatically generated and should go away from the repository.

Reply:
This file belongs to the gradle wrapper. The fact that it is automatically created isn't relevant here. The sole purpose is to reduce the complexity of the build process and to prevent potential problems with incompatible gradle versions.

Repositories that contain gradle wrapper scripts and configs do not require a manual gradle download. Users can simply run `./gradlew.sh assemble` to compile the code. If the required gradle version is not already installed it will be downloaded.

This is a well accepted procedure. You can read the documentation here: https://docs.gradle.org/current/userguide/gradle_wrapper.html

Reply:
Thank you for your expertise. I've read the documentation you linked to. I'm not Java developer but I don't agree with you. Automatically generated files don't have place in my repository. I've looked around and found this constructive discussion: https://stackoverflow.com/questions/20348451/why-should-the-gradle-wrapper-be-committed-to-vcs

Please read through it and propose a solution that suits you. I personally like the answer by Tomas Bjerre. I'm sure we can find a solution.

Reply:
From a theoretical point of view I understand your aversion against automatically generated files. But I think you are giving the automatic vs. non-automatic discussion way too much attention. In my opinion it should be a pragmatic discussion about pros vs. cons. The discussion you referenced is more about opinions than pragmatic solutions.

In particular, the comparison of the JDK (a **platform dependent** dependency that weights roughly **1GB** if you want your library to work on Linux, Mac and Windows, only counting x64 arch) and a **platform independent** dependency that is roughly **60-70KB** is very strange and misleading. As a pragmatic developer I always look at pros vs. cons. The cons of a nearly 1GB binary blob outweighs by far the benefit it could potentially have. A 70KB platform independent way to solve my dependency problems has certainly more pros than cons.

In his reply, Thomas asks

> "Do you also commit all your dependencie libraries?"

This is where he completely misses the intention behind the Gradle wrapper. It is not some random dependency. It bootstraps your build process. All of your other dependencies (including Gradle itself) are automatically downloaded. It's all about automation at the cost of just 70KB.

Installing Gradle manually and then calling the wrapper is possible. I did this for years and I had a rather negative experience with that approach. It is likely to introduce problems that haven't been addressed in the discussion. Not long ago, I got tons of questions related to [Could not determine java version from ‘9.0.1’](https://discuss.gradle.org/t/could-not-determine-java-version-from-9-0-1/24457) and similar problems. I learned my lesson ;)

Anyway, if you are still not convinced, just throw those files away and rebase the things you like into master. For me this is not a big deal. I will keep the wrapper in our fork since it works great for us.